pkg/controller/vsphere/reconciler.go

```diff
@@ -296,12 +297,21 @@ func (r *Reconciler) delete() error {
 		}
 	}
 
-	if _, err := vm.powerOffVM(); err != nil {
-		return err
+	powerState, err := vm.getPowerState()
+	if err != nil {
+		return fmt.Errorf("can not determine %v vm power state", r.machine.GetName())
+	}
+	if powerState != types.VirtualMachinePowerStatePoweredOff {
+		powerOffTaskRef, err := vm.powerOffVM()
+		if err != nil {
+			return fmt.Errorf("%v: failed to power off vm: %w", r.machine.GetName(), err)
+		}
+		if err := setProviderStatus(powerOffTaskRef, conditionSuccess(), r.machineScope, vm); err != nil {
+			return fmt.Errorf("Failed to set provider status: %w", err)
+		}
+		return fmt.Errorf("powering off vm is in progress, reconciling")
```

Comment:
Nit: For consistency with other messages
```suggestion
		return fmt.Errorf("powering off vm is in progress, requeuing")
```